### PR TITLE
fix inverted condition

### PR DESCRIPTION
**Pull request by matejvasek**
Fixing inverted condition, this caused really confusing error messages.

### CODE REVIEW

File: pkg/api/handlers/compat/images_push.go [hunk 1]
```diff
@@ -152,7 +152,7 @@ loop: // break out of for/select infinite loop
 		case err := <-pushErrChan:
 			if err != nil {
 				var msg string
-				if errors.Cause(err) != storage.ErrImageUnknown {
+				if errors.Is(err, storage.ErrImageUnknown) {
```
Comment (matejvasek) [inline pkg/api/handlers/compat/images_push.go:155]: Also I changed `Cause() ==` to `Is()` which is better for wrapped errors.

**Comment by rhatdan** [comment]
/approve
LGTM
@containers/podman-maintainers PTAL

**Review by Luap99 — APPROVED** [review]
LGTM

**Pull request merged**